surfactant/cmd/generate.py

```diff
@@ -29,7 +29,7 @@ def real_path_to_install_path(root_path: str, install_path: str, filepath: str)
 
 
 def get_software_entry(
-    context,
```

Comment:
current_context will need to be added as an argument for this function, then passed along to the hook call instead of None.